src/index.ts

```diff
@@ -30,6 +41,58 @@ if (program.endpointType === "client" && program.host === undefined) {
 
 const endpointType = program.endpointType;
 
+// If we are provided a config file, try reading it
+let hardwareConfig: IPinConfiguration[] | undefined = undefined;
+
+if (program.hardwareConfig !== undefined) {
+    try {
+        const portConfigsRaw: any = jsonfile.readFileSync(program.hardwareConfig);
+
+        // check if the raw object we get back is an array
+        if (portConfigsRaw instanceof Array) {
+            const portConfigs: string[] = (portConfigsRaw as string[]);
+            if (portConfigs.length < 5) {
```

Comment:
Why isn't this == 5? The for loop only goes up to 5 anyway so anything later would be unused.
Also, we should not use 5 and instead use const PORT_LENGTH or something.

Reply:
Yeah I can change that. Realistically, when we implement the code for the regular robot controller (the one we used for the original workshops), this should be a property of the controller type instead